### PR TITLE
Feature/save excel

### PR DESCRIPTION
**Pull request by netashefer**

### CODE REVIEW

File: src/helpers/xls-reader.ts [hunk 1]
```diff
@@ -19,4 +21,21 @@ export const convertToJson = (csv: string): Table => {
     }
 
     return { data: result, schema: headers };
+};
+
+export const onLoad = async (evt: ProgressEvent<FileReader>, file: Blob, callback: (dataSourceId: string, table: Table, info: any) => void) => {// evt = on_file_select event
+    const wb = getWorkBook(evt);
+    const wsname = wb.SheetNames[0];/* Get first worksheet */
```
Comment (YahelR) [inline src/helpers/xls-reader.ts]: Cool! Does this work the same way you know where? We take the data only from the first worksheet?

File: src/helpers/xls-reader.ts [hunk 2]
```diff
@@ -19,4 +21,21 @@ export const convertToJson = (csv: string): Table => {
...
+    const ws = wb.Sheets[wsname];
+    const data = XLSX.utils.sheet_to_csv(ws);/* Convert array of arrays */
+
+    const table: Table = convertToJson(data);
+    const dataSourceId = await excelCommunicator.addExcelDataSource({ table, dashboardId: "bla", displayName: "blas" });
```
Comment (YahelR) [inline src/helpers/xls-reader.ts:33]: do you want to generate the dashboardId in the ui or when inserting it to the db?

File: src/helpers/xls-reader.ts [hunk 3]
```diff
@@ -19,4 +21,21 @@ export const convertToJson = (csv: string): Table => {
     }
 
     return { data: result, schema: headers };
+};
+
+export const onLoad = async (evt: ProgressEvent<FileReader>, file: Blob, callback: (dataSourceId: string, table: Table, info: any) => void) => {// evt = on_file_select event
+    const wb = getWorkBook(evt);
```
Comment (YahelR) [inline src/helpers/xls-reader.ts]: can you change wb + ws to descriptive names? workBook + workSheet

File: src/helpers/xls-reader.ts (hunk 1 above)
Comment (YahelR) [inline src/helpers/xls-reader.ts]: workSheetName

Comment (netashefer) [reply src/helpers/xls-reader.ts]: it takes only the first

File: src/helpers/xls-reader.ts (hunk 2 above)
Comment (netashefer) [reply src/helpers/xls-reader.ts:33]: dashboardId will be known in the client (the server will generate it). also the dataSourceId the server genrates

**Pull request merged**